### PR TITLE
Add CI workflows and multi-session PR guard

### PR DESCRIPTION
**Pull request by jnasbyupgrade**
Implements the cross-repo CI design from pgxntool-test#15.

## What's added

**`.github/workflows/ci.yml`** — main CI workflow:
- Searches pgxntool-test for an open PR with the same branch name
- Waits up to 5 minutes for that PR's CI to pass before running tests
- Falls back to pgxntool-test/master if no test PR is found AND the `no-test-pr` label is set by a maintainer
- Fails (blocks merge) if neither condition is met

**`.github/workflows/protect-label.yml`** — label protection:
- Uses `pull_request_target` so it has write access from fork PRs
- Removes `no-test-pr` if added by a non-writer; re-adds it if removed by a non-writer
- Handles: non-collaborator 404s, bot-actor loop prevention, org team membership edge cases

**`.claude/CLAUDE.md`** — asks for confirmation before touching any existing PR not opened in this session (multiple concurrent Claude sessions are common across these repos)

## Setup required after merge
- Create the `no-test-pr` label in this repo (Settings → Labels)
- Add `resolve-test-ref` as a required status check on the `master` branch protection rule

Pairs with: https://github.com/Postgres-Extensions/pgxntool-test/pull/20

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by decibel** [comment]
@coderabbitai please review

**Comment by decibel** [comment]
@coderabbitai is there a way to configure you to post individual review comments using Github's Review Changes feature?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+    # (up to 20 minutes). The job timeout gives a few minutes of headroom.
+    timeout-minutes: 25
+    outputs:
+      run-tests: ${{ steps.check.outputs.run_tests }}
+      test-ref: ${{ steps.check.outputs.test_ref }}
+
+    steps:
+      - name: Find paired pgxntool-test PR or check commit-with-no-tests label
+        id: check
+        uses: actions/github-script@v7
```
[the remark this answers is not in the document]
Comment (jnasbyupgrade) [reply .github/workflows/ci.yml]: run-tests.yml is a known issue. Is there any significant risk to github-script @coderabbitai?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+            // (YAML expressions can't reference JS constants).
+            const NO_TEST_LABEL = 'commit-with-no-tests';
+
+            // master-to-master PRs have no paired test PR by convention.
+            // Run tests against pgxntool-test/master directly.
+            if (branch === 'master') {
+              core.setOutput('run_tests', 'true');
+              core.setOutput('test_ref', 'master');
+              return;
+            }
```
[the remark this answers is not in the document]
Comment (jnasbyupgrade) [reply .github/workflows/ci.yml:78]: Hrm... it's probably a mistake if someone commits to master on a fork. We shouldn't dis-allow it, but it'd be nice if CI posted a warning.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+            //
+            // The GitHub API's 'head' filter requires "owner:branch" format.
+            // We list all open PRs and filter locally — safe for repos with
+            // few open PRs, and avoids needing to know the fork repo name.
+            const { data: prs } = await github.rest.pulls.list({
+              owner: context.repo.owner,
+              repo: 'pgxntool-test',
+              state: 'open',
+              per_page: 100
+            });
```
[the remark this answers is not in the document]
Comment (jnasbyupgrade) [reply .github/workflows/ci.yml]: *very* unlikely this will ever happen, but we should have a comment explaining the risk. Bonus points if it spits out a very visible warning to the PR itself.

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jnasbyupgrade) [reply .github/workflows/ci.yml]: Ok, I think we don't need to do anything here.
